snuba/subscriptions/codecs.py

```diff
@@ -20,7 +21,14 @@ def encode(self, value: SubscriptionData) -> bytes:
         ).encode("utf-8")
 
     def decode(self, value: bytes) -> SubscriptionData:
-        data = json.loads(value.decode("utf-8"))
+        try:
+            data = json.loads(value.decode("utf-8"))
+        except json.JSONDecodeError:
+            raise InvalidQueryException("Invalid JSON")
+
+        if not data.get("aggregations"):
+            raise InvalidQueryException("No aggregation provided")
```

Comment:
Was this condition enforced anywhere else ?

Reply:
It wasn't. Either it fails on line 35 if the key "aggregations" is missing or it passes json schema validation (since it shares the same generic schema as a regular query where an aggregation is not mandatory) and then ClickHouse errors when running the query.

Reply:
Not following, Clickhouse should be able to execute a query without the aggregation field if it contains an aggregation function in the select.

Reply:
A subscription will never contain an aggregation function in the select, so unless there's at least one aggregation ClickHouse won't be able to execute the query

Reply:
Sorry I missed that you cannot add a select clause directly.